src/hotspot/share/classfile/systemDictionary.cpp

```diff
@@ -576,7 +577,7 @@ InstanceKlass* SystemDictionary::resolve_instance_class_or_null(Symbol* name,
                                                                 TRAPS) {
   // name must be in the form of "java/lang/Object" -- cannot be "Ljava/lang/Object;"
   assert(name != nullptr && !Signature::is_array(name) &&
-         !Signature::has_envelope(name), "invalid class name");
+         !Signature::has_envelope(name), "invalid class name: %s", name == nullptr ? "nullptr" : name->as_C_string());
```

Comment:
`name->as_C_String` requires a `ResourceMark` to be active - we probably should not assume/expect the caller has one.